pkg/api/v1/container.go

```diff
@@ -42,3 +47,26 @@ func (c *ContainerGroup) ListContainersByWorkspaceId(ctx echo.Context) error {
 
 	return ctx.JSON(http.StatusOK, containerStates)
 }
+
+func (c *ContainerGroup) StopAllWorkspaceContainers(ctx echo.Context) error {
+	workspaceId := ctx.Param("workspaceId")
+	containerStates, err := c.containerRepo.GetActiveContainersByWorkspaceId(workspaceId)
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, map[string]interface{}{
+			"message": "failed to stop containers",
+		})
+	}
+
+	for _, state := range containerStates {
+		err := c.scheduler.Stop(state.ContainerId)
```

Comment:
should we break on this error? or just continue to the rest of the containers

Reply:
Might be better to continue yea. I'll just log the error message